File: csharp/robot-name/RobotName.cs
using System;
using System.Collections.Generic;

public class Robot
{
    private static List<string> names = new List<string>();
    Random rand = new Random();
    public Robot()
    {
        Reset();
    }
    public string Name
    {
        get;
        set;
    }

    public void Reset()
    {
        do
        {
            Name = GetName();
        } while(names.Contains(Name));

        names.Add(Name);
    }

    private string GetRandChar()
    {
        string letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        return letters[rand.Next(letters.Length)].ToString();
    }

    private string GetRandInt()
    {

        const int MAX_NUM = 999;
        return rand.Next(MAX_NUM).ToString().PadLeft(3,'0');
    }

    private string GetName()
    {
        return GetRandChar() + 
                GetRandChar() + 
                GetRandInt();
    }
}

File: python/robot-name/robot_name.py
import secrets
import string

class Robot:

    names = []

    def __init__(self):
        self.name = None
        self.reset()

    def reset(self):
        self.name = self.get_name()
        Robot.names.append(self.name)

    def __create_name(self):
        letters = string.ascii_uppercase
        name = [secrets.choice(letters) for i in range(2)]
        name.append(str(secrets.randbelow(1000)).zfill(3))
        return ''.join(name)

    def get_name(self):
        temp_name = self.__create_name()

        while temp_name in Robot.names:
            temp_name = self.__create_name()

        return temp_name